### PR TITLE
fix:[PLG-352]AccountName population Issue for CQ Collector-GCP

### PR DESCRIPTION
**Pull request by AnandPaladin**
# Description

CQ Collector **does not contain AccountName** but AccountID. So we are fetching accountName using RDS from cf_Accounts table by passing accountId. We are using shipper module since it already contains code for RDS connection besides we also need to ship accountName to ES. So shipper module is apt one to make this changes.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Chore (no code changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Made shipper changes and ran the Shipper job for GCP using JobExecutor and checked ES data for accountName. Ensured these data are populated at SAASDEV as well as part of Paladin portal. Also tested negative test case of Azure Data -Shipper to make this RDS Call is only getting executed only for gcp-data-shipper but not for others.

Shipper_Job_Executor_Loca
OpenSearchResults_AccountName
Saasdev_AccName-Verification
Saasdev_Verification_Account_Name
Azure_Shipper_Job_Negative_Testing
AzureShipper_Negative_Test_Case

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] My commit message/PR follows the contribution guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# **Other information**:

List any documentation updates that are needed for the Wiki

### CODE REVIEW

File: jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/EntityManager.java [hunk 1]
```diff
@@ -262,7 +264,32 @@ private void prepareDocs(Map<String, Map<String, String>> currentInfo, List<Map<
             } else if (entityInfo.containsKey("projectId")) {
                 entityInfo.put("accountid", entityInfo.get("projectId"));
             }
+            //For GCP CQ Collector accountName will be fetched from RDS using accountId
```
Comment (plyubich) [inline jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/EntityManager.java:267]: @AnandPaladin we would know that data source at the beginning of the job. Let's cache the DB account name request once at the beginning of the job.

**Review by plyubich — CHANGES_REQUESTED** [review]
Implement caching of the GCP account name

File: jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/EntityManager.java (hunk 1 above)
Comment (AnandPaladin) [reply jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/entity/EntityManager.java:267]: @plyubich I have used HashMap which act as **InMemorycache** to check whether it already contains {projectId,project Name} .Only if it does not contain entry then only we are making RDS call. Please clarify this is not serving the purpose as that of cache ?

CC:@mkinchin-paladin

**Review by plyubich — APPROVED** [review]
